### PR TITLE
Add component feature for plotting vectors

### PR DESCRIPTION
**Pull request by banesullivan**
Continuation of #543 

pinging @MatthewFlamm 

This fixes the array title issues.

Also, note that we used to forcibly copy/overwrite existing scalars and these changes prevent scalars that already exist on a mesh from being copied, which should speed things up a bit when plotting larger meshes (maybe?).

### CODE REVIEW

File: examples/02-plot/vector-component.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+###############################################################################
+
+dargs = dict(scalars="gradient", cmap="rainbow")
+
+p = pv.Plotter(shape=(1,3))
+p.subplot(0,0)
+p.add_mesh(mesh, component=0, **dargs)
+p.add_text("Component 0")
+p.subplot(0,1)
+p.add_mesh(mesh.copy(), component=1, **dargs)
```
Comment (MatthewFlamm) [inline examples/02-plot/vector-component.py]: Should it be pointed out why `mesh.copy()` is needed here?

File: tests/test_plotting.py [hunk 2]
```diff
@@ -926,3 +926,93 @@ def test_default_name_tracking():
...
+def test_vector_array_fail_with_incorrect_component():
+    data = examples.load_uniform()
+
+    vector_values_points = np.empty((data.n_points, 3))
+    vector_values_points[:, :] = [3., 4., 0.]  # Vector has this value at all points
+
+    data['vector_values_points'] = vector_values_points
+
+    p = pyvista.Plotter(off_screen=OFF_SCREEN)
+
```
Comment (MatthewFlamm) [inline tests/test_plotting.py]:
```suggestion
```

**Comment by MatthewFlamm** [comment]
This works well on my end, but I only tested the vector piece of it.  It looks like the scalar bar title behavior makes slightly more sense.

I still don't like 'vector-valued scalars' in the docstring even though I wrote it.

**Comment by akaszynski** [comment]
@banesullivan, this is an old PR, but I still like the idea of being able to plot components of scalar arrays.  Think it's possible to have this done at some point?

**Comment by banesullivan** [comment]
Possible? Yes! We may want to start a new PR with this...

Unfortunately, I don't know if I will have the time anytime soon 😞

**Comment by MatthewFlamm** [comment]
I propose, as a temporary measure, that we drop titling the scalar bar according to whether it is a normed vector or a component of a vector.  This is no worse than the current titling state, which is currently broken for vectors.  The reason that this PR is complex, IMO, is the complexity of the scalar bar titling logic, not the plotting of the component itself.  I cannot wrap my head around the logic of the scalar bar titling as it morphs throughout this function.

If this seems OK as a path forward @banesullivan , I can submit a new PR with just this piece.

**Comment by banesullivan** [comment]
> If this seems OK as a path forward @banesullivan , I can submit a new PR with just this piece.

I'm all for that, @MatthewFlamm, go ahead! APologies I haven't had the bandwidth to come back to this

**Pull request closed without merging**